core/lib/spree/core/controller_helpers/auth.rb

```diff
@@ -28,6 +29,17 @@ module Auth
           rescue_from CanCan::AccessDenied do
             instance_exec(&unauthorized_redirect)
           end
+
+          # Triggers after devise has logged a user in.
+          def after_sign_in_path_for(resource)
```

Comment:
does this happen only when a user logs in via web app?

Reply:
Correct
see: https://www.rubydoc.info/github/plataformatec/devise/Devise/Controllers/Helpers:after_sign_in_path_for

When a user has logged in through the Devise controller this method is called.